src/uu/tee/src/tee.rs

```diff
@@ -5,88 +5,73 @@
+        .arg(
+            Arg::with_name(options::APPEND)
+                .long(options::APPEND)
+                .short("a")
+                .help("append to the given FILEs, do not overwrite"),
+        )
+        .arg(
+            Arg::with_name(options::IGNORE_INTERRUPTS)
+                .long(options::IGNORE_INTERRUPTS)
+                .short("i")
+                .help("ignore interrupt signals (ignored on non-Unix platforms)"),
+        )
+        .arg(Arg::with_name(options::FILE).multiple(true))
+        .get_matches_from(args);
+
+    let options = Options {
+        append: matches.is_present(options::APPEND),
+        ignore_interrupts: matches.is_present(options::IGNORE_INTERRUPTS),
+        files: matches
+            .values_of(options::FILE)
+            .map(|v| v.map(ToString::to_string).collect())
+            .unwrap_or_default(),
+    };
 
-    match options(&args).and_then(exec) {
+    match tee(options) {
         Ok(_) => 0,
         Err(_) => 1,
     }
 }
 
 #[allow(dead_code)]
 struct Options {
-    program: String,
     append: bool,
     ignore_interrupts: bool,
-    print_and_exit: Option<String>,
     files: Vec<String>,
 }
 
-fn options(args: &[String]) -> Result<Options> {
-    let mut opts = getopts::Options::new();
-
-    opts.optflag("a", "append", "append to the given FILEs, do not overwrite");
-    opts.optflag(
-        "i",
-        "ignore-interrupts",
-        "ignore interrupt signals (ignored on non-Unix platforms)",
-    );
-    opts.optflag("h", "help", "display this help and exit");
-    opts.optflag("V", "version", "output version information and exit");
-
-    opts.parse(&args[1..])
-        .map_err(|e| Error::new(ErrorKind::Other, format!("{}", e)))
-        .map(|m| {
-            let version = format!("{} {}", NAME, VERSION);
-            let arguments = "[OPTION]... [FILE]...";
-            let brief = "Copy standard input to each FILE, and also to standard output.";
-            let comment = "If a FILE is -, it refers to a file named - .";
-            let help = format!(
-                "{}\n\nUsage:\n  {} {}\n\n{}\n{}",
-                version,
-                NAME,
-                arguments,
-                opts.usage(brief),
-                comment
-            );
-            let names: Vec<String> = m.free.clone().into_iter().collect();
-            let to_print = if m.opt_present("help") {
-                Some(help)
-            } else if m.opt_present("version") {
-                Some(version)
-            } else {
-                None
-            };
-            Options {
-                program: NAME.to_owned(),
-                append: m.opt_present("append"),
-                ignore_interrupts: m.opt_present("ignore-interrupts"),
-                print_and_exit: to_print,
-                files: names,
-            }
-        })
-        .map_err(|message| warn(format!("{}", message).as_ref()))
+mod options {
```

Comment:
could you please move it above the main ?
(for consistency with other programs)
thanks

Reply:
done